File: app/src/main/java/work/petrichor/petrichor/modules/GoogleFaceDetectionListener.java
package work.petrichor.petrichor.modules;

import android.hardware.Camera;
import android.os.Handler;
import android.os.Message;

/**
 * @author Mitscherlich.
 * @date 2017/7/27.
 **/

public class GoogleFaceDetectionListener implements Camera.FaceDetectionListener {

    private Handler mHandler;

    public GoogleFaceDetectionListener(Handler mHandler) {
        this.mHandler = mHandler;
    }

    @Override
    public void onFaceDetection(Camera.Face[] faces, Camera camera) {
        Message message = mHandler.obtainMessage();
        message.what = CameraInterface.ON_GOOGLE_FACE_DETECTION_MESSAGE;
        message.obj = faces;
        message.sendToTarget();
    }
}

File: app/src/main/java/work/petrichor/petrichor/modules/GoogleFaceDetectionListener.kt
package work.petrichor.petrichor.modules

import android.hardware.Camera
import android.os.Handler

/**
 * @author  Mitscherlich.
 * @date    2017/7/24.
 * **/
class GoogleFaceDetectionListener(private val handler: Handler) : Camera.FaceDetectionListener {

    override fun onFaceDetection(faces: Array<out Camera.Face>?, camera: Camera?) {
        val message = handler.obtainMessage()
        message.what = CameraInterface.ON_GOOGLE_FACE_DETECTION_MESSAGE
        message.obj = faces ?: emptyArray<Camera.Face>()
        message.sendToTarget()
    }
}